From 6aa4a3eac1a2be62da89ef3316aa53b78588b8bc Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Thu, 8 Mar 2012 21:56:11 +0000
Subject: [PATCH 504/507] mageia: Fallback message when display manager fails.

Provide a unit for displaying a message to the user when the
display manager fails. This can be when no dm application is
installed or when it exits with a failure code.

Much like the prefdm script, the actual message is included
in the initscripts package.

https://bugs.mageia.org/show_bug.cgi?id=4769

v2: Use agetty rather than mingetty + quit plymouth + don't wait for
    graphical.target
v3: Use Type=idle now available upstream
---
 Makefile.am                                  |    1 +
 units/mageia/display-manager-failure.service |   30 ++++++++++++++++++++++++++
 units/mageia/prefdm.service                  |    1 +
 3 files changed, 32 insertions(+)
 create mode 100644 units/mageia/display-manager-failure.service

diff --git a/Makefile.am b/Makefile.am
index 1b61b8c..4cc20f2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -419,6 +419,7 @@ endif
 if TARGET_MAGEIA
 dist_systemunit_DATA += \
 	units/mageia/prefdm.service \
+	units/mageia/display-manager-failure.service \
 	units/fedora/rc-local.service \
 	units/fedora/halt-local.service
 systemgenerator_PROGRAMS += \
diff --git a/units/mageia/display-manager-failure.service b/units/mageia/display-manager-failure.service
new file mode 100644
index 0000000..3a79ba5
--- /dev/null
+++ b/units/mageia/display-manager-failure.service
@@ -0,0 +1,30 @@
+[Unit]
+Description=Display Manager Failure Message
+After=dev-tty1.device systemd-user-sessions.service shutdown.target
+
+Conflicts=getty@tty1.service
+After=getty@tty1.service
+
+[Service]
+Environment=TERM=linux
+ExecStartPre=-/bin/plymouth quit
+ExecStart=-/sbin/agetty --noissue --nohostname -n --autologin nobody --login-program /usr/bin/display-manager-failure-message tty1
+Type=idle
+Restart=always
+RestartSec=0
+UtmpIdentifier=tty1
+TTYPath=/dev/tty1
+TTYReset=yes
+TTYVHangup=yes
+TTYVTDisallocate=yes
+KillMode=process
+IgnoreSIGPIPE=no
+
+# Unset locale for the console getty since the console has problems
+# displaying some internationalized messages.
+Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= LC_MESSAGES= LC_PAPER= LC_NAME= LC_ADDRESS= LC_TELEPHONE= LC_MEASUREMENT= LC_IDENTIFICATION=
+
+# Some login implementations ignore SIGTERM, so we send SIGHUP
+# instead, to ensure that login terminates cleanly.
+KillSignal=SIGHUP
+
diff --git a/units/mageia/prefdm.service b/units/mageia/prefdm.service
index db41294..84990b4 100644
--- a/units/mageia/prefdm.service
+++ b/units/mageia/prefdm.service
@@ -14,6 +14,7 @@ After=livesys-late.service systemd-user-sessions.service
 # Do not stop plymouth, it is done in prefdm if required (or left to the dm)
 Conflicts=getty@tty1.service plymouth-quit.service
 After=getty@tty1.service plymouth-quit.service
+OnFailure=display-manager-failure.service
 
 [Service]
 ExecStart=/etc/X11/prefdm -nodaemon
-- 
1.7.10.3

